models/Funding.php

```diff
@@ -312,4 +363,58 @@ public function getGallery()
 
         return $gallery;
     }
+
+    private function AttachSpace()
+    {
+        /* @var Module $module */
+        $module = Yii::$app->getModule('space');
+
+        // init space model
+        $space = new Space();
+        $space->scenario = Space::SCENARIO_CREATE;
+        $space->visibility = $module->settings->get('defaultVisibility', Space::VISIBILITY_REGISTERED_ONLY);
+        $space->join_policy = $module->settings->get('defaultJoinPolicy', Space::JOIN_POLICY_APPLICATION);
+        $space->color = RandomColor::one(['luminosity' => 'dark']);
+        $space->space_type = Space::SPACE_TYPE_FUNDING;
+        $space->name = $this->title;
+        $space->description = $this->description;
+        $space->url = UrlValidator::autogenerateUniqueSpaceUrl($this->title);
+
+        if (!$space->save()) {
+            throw new HttpException(400);
+        }
+
+        $this->space_id = $space->id;
+    }
+
+    private function createFundingAccount()
+    {
+        $account = new Account();
+        $account->space_id = $this->space->id;
+        $account->title = "Campaign # $this->id";
+        $account->account_type = Account::TYPE_FUNDING;
+        $account->funding_id = $this->id;
+
+        if (!$account->save()) {
+            throw new \yii\base\Exception('Could not create funding account!');
```

Comment:
translation here, also either use `\yii\base\Exception()` or `Exception()`

Reply:
we're not using translation for DB errors